src/Avalonia.Base/Media/Imaging/RenderTargetBitmap.cs

```diff
@@ -61,6 +61,25 @@ private static IRenderTargetBitmapImpl CreateImpl(PixelSize size, Vector dpi)
         /// <inheritdoc/>
         public IDrawingContextImpl CreateDrawingContext(IVisualBrushRenderer? vbr) => PlatformImpl.Item.CreateDrawingContext(vbr);
 
+        /// <inheritdoc/>
         bool IRenderTarget.IsCorrupted => false;
+
+        /// <summary>
+        /// Overrides the <see cref="Bitmap"/> implementation 
+        /// skipping the <see cref="Bitmap.Format"/> null check
+        /// </summary>
+        public override void CopyPixels(PixelRect sourceRect, nint buffer, int bufferSize, int stride)
+        {
+            // only check if we have a readable bitmap and if the formats are identical
+            // this also allows for both formats being null
+            if (PlatformImpl.Item is not IReadableBitmapImpl readable
+                || Format != readable.Format
```

Comment:
The Format is still `null` though. How exactly the user code is supposed to allocate the buffer if it doesn't know the pixel size?

Reply:
Good point! Why does it have Format == null, because the format only kind of exists when you create the snapshot from the render target. Which is why I would not put RenderTargetBitmap.Format => Snapshot().Format that is not like a typical use of a property. I would see the option to provide a RenderTargetBitmap.PeekFormat() or similar which makes clear to the user that this is not a no-cost call. What says you?

Reply:
Like this:
```csharp
        public PixelFormat? PeekFormat()
        {
            if (PlatformImpl.Item is not IReadableBitmapImpl readable)
                return null;

            using (var fb = readable.Lock())
                return fb.Format;
        }
```

Reply:
Alternatively, we could set RenderTargetBitmap.Format to fb.Format but use a sort of caching sucht that the FB.Lock is only called once.

Reply:
While trying it seems to me that an efficient ```IReadableBitmapImpl.Format``` implementation in ```SurfaceRenderTarget``` may be the better option. So the client does not have the obligation to call PeekFormat once but can use just the "normal" Format. Special about this is that the Format is only provided by the SnaptshotImage() call, hence, needs to be cached to avoid such repeated calls.